### PR TITLE
Azure AD Authentication with Next.js and Prisma

### PR DESCRIPTION
**Pull request by mvthul**
Al that is needed are the following environment variables: 
AZURE_AD_CLIENT_ID=
AZURE_AD_CLIENT_SECRET=
AZURE_AD_TENANT_ID=
ALLOW_DANGEROUS_EMAIL_LINKING=true (Somehow it doesnt work without this) 
OAUTH_EXCLUSIVE_LOGIN=true (Shows only Sign in with Azure AD button)

### CODE REVIEW

**Comment by mvthul** [comment]
Al that is needed are the following environment variables:
AZURE_AD_CLIENT_ID=
AZURE_AD_CLIENT_SECRET=
AZURE_AD_TENANT_ID=
ALLOW_DANGEROUS_EMAIL_LINKING=true (Somehow it doesnt work without this)
OAUTH_EXCLUSIVE_LOGIN=true (Shows only Sign in with Azure AD button)

File: .devcontainer/init-cmd.sh [hunk 1]
```diff
@@ -26,6 +26,7 @@ chmod +x /usr/local/bin/ztmkworld
 # apply migrations to the database
 echo "Applying migrations to the database..."
 npx prisma migrate deploy
+# npm install && npm run dev
```
Comment (sinamics) [inline .devcontainer/init-cmd.sh]: remove this line

File: prisma/migrations/20230402192938_/migration.sql [hunk 2]
```diff
@@ -59,6 +59,8 @@ CREATE TABLE "Account" (
     "refresh_token" TEXT,
     "access_token" TEXT,
     "expires_at" INTEGER,
+    "expires_in" INTEGER,
+    "ext_expires_in" INTEGER,
     "token_type" TEXT,
```
Comment (sinamics) [inline prisma/migrations/20230402192938_/migration.sql:64]: are these really neccessary?

File: src/server/auth.ts [hunk 3]
```diff
@@ -107,6 +108,37 @@ export const authOptions: NextAuthOptions = {
...
+				return Promise.resolve({
+					id: profile.sub || profile.id.toString(), // Handle ID based on provider
+					name: profile.name || profile.login || profile.username,
+					email: profile.email,
+					image: profile.picture || profile.avatar_url || profile.image_url || profile.image,
+					role: "USER",
+				});
+			},
+		}),
 		CredentialsProvider({
```
Comment (sinamics) [inline src/server/auth.ts]: remove all unused code.
also, i would prefer if we could incorporate this into the custom oauth adapter. I dont want to add code for all providers out there.

**Review by sinamics — COMMENTED** [review]
Documentation needs to be updated as well.

File: prisma/migrations/20230402192938_/migration.sql (hunk 2 above)
Comment (mvthul) [reply prisma/migrations/20230402192938_/migration.sql:64]: > are these really neccessary?

I can try without but seems necessary according to documentation next auth.

File: .devcontainer/init-cmd.sh (hunk 1 above)
Comment (mvthul) [inline .devcontainer/init-cmd.sh]: Done

**Review by mvthul — COMMENTED** [review]
I updated the files

**Comment by sinamics** [comment]
Added azure oauth to the generic provider. #483

**Pull request closed without merging**